### PR TITLE
JBPM-6023 - HTTP Code 500 is returned when starting process with non-…

### PR DESCRIPTION
**Pull request by tsurdilo**
…existing process ID or deployment ID

### CODE REVIEW

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/bpmn2/BPMN2DataServiceImpl.java [hunk 1]
```diff
@@ -190,18 +191,16 @@ private ProcessAssetDesc fillProcessDefinition(ProcessDescriptor helper, KieCont
...
-	        return helper.getProcess();
-	    }
-
-	    return null;
+        if (definitionCache.containsKey(deploymentId)) {
+            ProcessDescriptor helper = definitionCache.get(deploymentId).get(processId);
+            if(helper != null && helper.getProcess() != null) {
+                return helper.getProcess();
+            } else {
+                throw new ProcessDefinitionNotFoundException("No process available with given id : " + processId);
```
Comment (mswiderski) [inline jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/bpmn2/BPMN2DataServiceImpl.java]: this requires change in interface description https://github.com/kiegroup/jbpm/blob/master/jbpm-services/jbpm-services-api/src/main/java/org/jbpm/services/api/DefinitionService.java#L62 as it won't return null any more as it is stated there

**Comment by MarianMacik** [comment]
Looks good. Once the interface documentation is updated, this can be merged.

**Comment by tsurdilo** [comment]
@mswiderski @MarianMacik amended - thanks!

**Comment by tsurdilo** [comment]
@mswiderski @MarianMacik can me merged if approved now i think. thanks

**Pull request merged**